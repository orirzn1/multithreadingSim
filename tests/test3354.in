L14
S8
O12
N17

T0
I@0x00000000
STORE $7, $0, $4
SUBI $5, $1, 0x1
STORE $5, $5, 0x1
SUBI $0, $1, 0x2
ADDI $1, $5, 0x3
LOAD $6, $2, $3
ADD $3, $4, $5
ADDI $4, $2, 0x1
LOAD $5, $6, 0x1
ADDI $0, $6, 0x4
HALT $0

T1
I@0x00000000
ADD $1, $3, $2
SUB $2, $2, $1
STORE $6, $3, $3
STORE $4, $4, 0x0
SUB $7, $2, $6
HALT $0

T2
I@0x00000000
SUBI $6, $1, 0x3
ADD $6, $6, $5
ADD $0, $3, $2
ADDI $3, $5, 0x3
LOAD $4, $4, $3
LOAD $4, $0, $4
SUBI $7, $6, 0x2
SUBI $2, $6, 0x2
HALT $0

T3
I@0x00000000
ADDI $7, $0, 0x4
ADD $6, $4, $0
SUBI $4, $7, 0x1
SUBI $3, $2, 0x2
HALT $0

T4
I@0x00000000
ADD $1, $7, $4
ADDI $6, $4, 0x1
SUB $4, $4, $7
STORE $4, $5, 0x0
SUB $5, $7, $4
SUBI $7, $7, 0x4
SUBI $2, $0, 0x2
LOAD $7, $5, $6
STORE $2, $4, 0x1
HALT $0

T5
I@0x00000000
SUB $7, $4, $5
ADD $4, $7, $5
STORE $2, $2, $3
SUBI $0, $1, 0x4
ADD $2, $4, $3
STORE $1, $3, 0x3
HALT $0

T6
I@0x00000000
SUB $3, $6, $0
ADD $7, $5, $4
SUBI $6, $0, 0x2
ADDI $0, $7, 0x2
HALT $0

T7
I@0x00000000
LOAD $0, $4, 0x2
ADDI $2, $6, 0x1
STORE $0, $1, 0x3
ADDI $4, $2, 0x1
ADDI $6, $2, 0x4
SUBI $1, $6, 0x1
SUB $3, $3, $0
HALT $0

T8
I@0x00000000
SUBI $3, $5, 0x2
SUB $0, $2, $7
SUB $3, $1, $7
SUB $2, $4, $5
ADD $6, $3, $4
ADDI $3, $0, 0x1
HALT $0

T9
I@0x00000000
ADD $6, $4, $5
LOAD $5, $3, $4
LOAD $5, $0, 0x2
LOAD $4, $7, $0
SUB $4, $5, $6
ADD $6, $6, $6
LOAD $0, $0, $4
SUBI $6, $6, 0x4
SUBI $7, $6, 0x1
LOAD $0, $4, $0
HALT $0

T10
I@0x00000000
LOAD $5, $3, $1
ADD $5, $7, $1
STORE $6, $3, 0x1
STORE $5, $1, 0x2
ADDI $6, $5, 0x1
ADDI $7, $5, 0x2
SUB $5, $5, $7
SUBI $4, $5, 0x1
SUBI $4, $2, 0x3
STORE $7, $3, 0x0
HALT $0

T11
I@0x00000000
STORE $3, $4, 0x0
SUB $3, $6, $1
LOAD $5, $7, $3
SUBI $2, $7, 0x2
ADD $3, $2, $4
SUB $4, $3, $3
STORE $3, $1, 0x0
STORE $3, $2, $4
ADD $0, $6, $2
HALT $0

T12
I@0x00000000
STORE $2, $6, $2
SUB $5, $2, $1
STORE $6, $5, $7
ADD $4, $1, $7
ADDI $7, $7, 0x2
SUBI $6, $5, 0x2
HALT $0

T13
I@0x00000000
LOAD $2, $1, $2
SUB $1, $2, $0
SUBI $6, $5, 0x1
ADDI $1, $1, 0x1
LOAD $5, $5, $3
ADD $1, $2, $3
HALT $0

T14
I@0x00000000
LOAD $4, $3, 0x2
LOAD $4, $5, $3
ADDI $1, $1, 0x1
SUB $0, $1, $4
SUBI $5, $5, 0x3
HALT $0

T15
I@0x00000000
ADD $4, $4, $5
ADD $4, $2, $5
ADD $1, $6, $1
ADD $4, $6, $0
SUB $3, $6, $7
SUB $3, $1, $3
HALT $0

T16
I@0x00000000
SUBI $1, $1, 0x1
SUBI $1, $1, 0x1
STORE $2, $0, $0
ADDI $6, $2, 0x1
ADD $0, $2, $2
LOAD $6, $2, 0x4
ADD $5, $1, $3
ADD $4, $4, $4
ADD $0, $3, $1
ADDI $1, $6, 0x3
HALT $0

D@0x0
0x3
0x3
0x2
0x1
0x7
0x7
