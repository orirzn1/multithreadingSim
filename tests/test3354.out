
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x4	R1 = 0x2	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0x1	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFB	R3 = 0x3	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0xFFFFFFFB
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFE	R4 = 0x3	R5 = 0x0	R6 = 0x0	R7 = 0x4
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x1	R7 = 0x0
Register file thread id 5:
	R0 = 0xFFFFFFFC	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x2	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0x4	R2 = 0x1	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0x5	R7 = 0x0
Register file thread id 8:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 9:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0xFFFFFFFB
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0xFFFFFFFE	R6 = 0x1	R7 = 0x2
Register file thread id 11:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 12:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x2
Register file thread id 13:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0x0
Register file thread id 14:
	R0 = 0x1	R1 = 0x1	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFD	R6 = 0x0	R7 = 0x0
Register file thread id 15:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 16:
	R0 = 0xFFFFFFFE	R1 = 0x6	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0x3	R7 = 0x0
Blocked MT CPI for this program 5.521739

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x4	R1 = 0x2	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0x1	R5 = 0xFFFFFFFE	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFB	R3 = 0x3	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0xFFFFFFFB
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFE	R4 = 0x3	R5 = 0x0	R6 = 0x0	R7 = 0x4
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x1	R7 = 0x0
Register file thread id 5:
	R0 = 0xFFFFFFFC	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x2	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0x4	R2 = 0x1	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0x5	R7 = 0x0
Register file thread id 8:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0xFFFFFFFB
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0xFFFFFFFE	R6 = 0x1	R7 = 0x2
Register file thread id 11:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 12:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x2
Register file thread id 13:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0x0
Register file thread id 14:
	R0 = 0x1	R1 = 0x1	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFD	R6 = 0x0	R7 = 0x0
Register file thread id 15:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 16:
	R0 = 0xFFFFFFFE	R1 = 0x6	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0x3	R7 = 0x0
Finegrained Multithreading CPI for this program 1.050725

